### ISSUE TITLE
Issuing adb command crashes appium

### ISSUE THREAD

**Issue by adamf663** [appium/appium#6537]
If an adb command is issued, appium crashes.  

Why won't appium fix this?  This is the third time I'm reporting this.  I originally thought the issue was trying to use two instances of appium as appium would crash almost right away when there were two instances.  

The problem occurs when an adb command is issued.    When I posted this as a separate bug, it was marked as a duplicate of the multiple appium server crash and closed.

This bug makes appium unuseable.  You really ought to fix it.

**Comment by mykola-mokhnach** [comment]
Dear @adamf663 
please be kind to carefully provide all the necessary information, which is required in the standard issue template. It is practically impossible to solve anything without logs, steps, screenshots, - everything that can help to investigate an issue.

**Comment by adamf663** [comment]
They were all provided in https://github.com/appium/appium/issues/6384
I tire of reentering it after having it closed without any work being done.

----- Original Message -----

**Comment by mykola-mokhnach** [comment]
We can try to make a workaround for this issue. Probably, you can just take a screenshot directly, not using appium for this purpose:
```java
    private static BufferedImage fixScreenshotOrientation(final BufferedImage initialScreenshot,
                                                          SurfaceOrientation currentOrientation) throws IOException {
        if (currentOrientation != SurfaceOrientation.ROTATION_0) {
            return ImageUtil.tilt(initialScreenshot, -currentOrientation.getCode() * Math.PI / 2);
        }
        return initialScreenshot;
    }

    public static void takeAndroidScreenshot(ZetaAndroidDriver driver, File resultScreenshot, boolean shouldApplyScale)
            throws Exception {
        final String pathOnPhone = String.format("/sdcard/%s.png", generateGUID().replace("-", "").substring(0, 8));
        final String adbCommandsChain = String.format(ADB_PREFIX + "adb shell screencap -p %1$s; " + ADB_PREFIX
                        + "adb pull %1$s %2$s; " + ADB_PREFIX + "adb shell rm %1$s", pathOnPhone,
                resultScreenshot.getCanonicalPath());
        Runtime.getRuntime().exec(new String[]{"/bin/bash", "-c", adbCommandsChain}).waitFor();
        byte[] output = FileUtils.readFileToByteArray(resultScreenshot);
        final boolean isTablet = getIsTabletFromConfig(CommonUtils.class);
        BufferedImage resultImg = ImageIO.read(new ByteArrayInputStream(output));
        if (isTablet) {
            SurfaceOrientation currentOrientation = SurfaceOrientation.ROTATION_0;
            try {
                currentOrientation = driver.getSurfaceOrientation();
            } catch (Exception e) {
                e.printStackTrace();
            }
            log.debug(String.format("Current screen orientation value -> %s", currentOrientation.getCode()));
            resultImg = fixScreenshotOrientation(resultImg, currentOrientation);
            if (shouldApplyScale) {
                resultImg = ImageUtil.scaleTo(resultImg,
                        MAX_TABLET_SCREENSHOT_WIDTH, MAX_TABLET_SCREENSHOT_HEIGHT);
            }
        } else {
            if (shouldApplyScale) {
                resultImg = ImageUtil.scaleTo(resultImg,
                        MAX_PHONE_SCREENSHOT_WIDTH, MAX_PHONE_SCREENSHOT_HEIGHT);
            }
        }
        ImageUtil.storeImage(resultImg, resultScreenshot);
    }
```

**Comment by adamf663** [comment]
The problem is that any adb command issued from anywhere will crash appium.  All I have to do to crash appium is run "watch adb devices" in a window.


----- Original Message -----

**Comment by mykola-mokhnach** [comment]
Yes, I understand it. It might be possible, that Appium executes adb and then reads the stream (actually, this is where (E)PIPE is coming from). Manual adb command execution breaks this stream and that is why we have Appium crash as a result. The trick is to avoid Appium issuing commands, that require to lock adb stream, and use custom stuff instead.

**Comment by adamf663** [comment]
I would be nice if there was more diagnostic information other than "appium has crashed [EPIPE]".  I've asked about a half dozen times for *any* additional information and have currently given up;  time permitting, I intend to rip appium out of our automation framework.  Will use uiautomator directly, move test execution to a "primary" phone, and implement a servier/client architecture at the page object level to remotely control other phones. 

----- Original Message -----

**Comment by adamf663** [comment]
This has been fixed in 1.5.2 somehow.  "Only" two months downtime.  It would have been nice is 1) there was more diagnostic information than "EPIPE", 2) there had been a workaround so I wouldn't have had my entire automation framework down for two months and 3) the work had been done on any of the multiple issues I entered instead of them being closed out without being fixed and the fix being quietly slipped in probably without a ticket.
